### PR TITLE
gh-120129: Slices length equals to the difference of the indices

### PR DESCRIPTION
**Pull request by SweetyAngel**
```the length of a slice is the difference of the indices``` works with negative indices also.
Look in issue

* Issue: gh-120129

----
📚 Documentation preview 📚: https://cpython-previews--120130.org.readthedocs.build/

### CODE REVIEW

File: Doc/tutorial/introduction.rst [hunk 1]
```diff
@@ -315,9 +315,9 @@ the second row gives the corresponding negative indices. The slice from *i* to
 *j* consists of all characters between the edges labeled *i* and *j*,
 respectively.
 
-For non-negative indices, the length of a slice is the difference of the
-indices, if both are within bounds.  For example, the length of ``word[1:3]`` is
-2.
+The length of a slice is the difference of the
```
Comment (sobolevn) [inline Doc/tutorial/introduction.rst:318]: It is more complex than that. Please, take a look at this example:

```python
>>> len(range(2, -2, -1))
4
```

Source: https://github.com/python/cpython/blob/f878d46e5614f08a9302fcb6fc611ef49e9acf2f/Objects/rangeobject.c#L224

**Review by sobolevn — COMMENTED** [review]
I agree that the docs should be updated, but we need to tweak the wording a bit more.

File: Doc/tutorial/introduction.rst (hunk 1 above)
Comment (SweetyAngel) [reply Doc/tutorial/introduction.rst:318]: But this part is about slices of strings, not range object. We can talk about this example in range reference

Comment (SweetyAngel) [reply Doc/tutorial/introduction.rst:318]: I.e: length of range object equals ```math.ceil((stop - start) / step)```

Comment (sobolevn) [reply Doc/tutorial/introduction.rst:318]: oh, sorry, it was too late for me to do a review 😆

File: Doc/tutorial/introduction.rst [hunk 2]
```diff
@@ -315,9 +315,9 @@ the second row gives the corresponding negative indices. The slice from *i* to
...
+indices, if both are within bounds and their signs coincide.  For example, the length of ``word[1:3]`` is
+2 and the length of ``word[-4:-1]`` is -1-(-4) = 3.
```
Comment (sobolevn) [inline Doc/tutorial/introduction.rst:320]: What about 

```python
>>> 'wolrds'[3:1:-1]
'rl'
```

do we care about this example?

Comment (SweetyAngel) [reply Doc/tutorial/introduction.rst:320]: Do you know where the ```step``` parameter of slice is mentioned? I cant find it

Comment (SweetyAngel) [reply Doc/tutorial/introduction.rst:320]: If there is no information about that, we need to add it

Comment (SweetyAngel) [reply Doc/tutorial/introduction.rst:320]: I found something similar but not quite https://docs.python.org/3/library/functions.html#slice

**Comment by rhettinger** [comment]
I recommend leaving this as-is.  In an introduction, we need to cover common cases and help the reader form a basic mental model.

At this point in the tutorial, it makes users worse off to explore "second-level" knowledge before they've achieved "first-level" understanding.

IIRC, we used similar reasoning to reject an edit going in the details of `int.__mod__` in the presence of negative numbers. A user who cared could easily reason it out or do a quick exploration at an interactive prompt.  Users who don't care (most of them) wouldn't be burdened by esoterica.  At this point in the tutorial, they've still got a long way to go.

In the case of slicing, we've never had user report an issue on the slice arithmetic explanation in the 25-year history of this text (at least not with regard to negative indices and the slice length).

**Comment by sobolevn** [comment]
I agree

**Pull request closed without merging**